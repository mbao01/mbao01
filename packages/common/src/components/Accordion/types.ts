import * as AccordionPrimitive from "@radix-ui/react-accordion";

export type AccordionProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Root>;

export type AccordionContentProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Content
>;

export type AccordionItemProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Item>;

export type AccordionTriggerProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Trigger
>;
