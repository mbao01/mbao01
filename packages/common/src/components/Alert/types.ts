import { type VariantProps } from "../../libs";
import { getAlertClasses } from "./constants";

export type AlertProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getAlertClasses>;

export type AlertTitleProps = React.HTMLAttributes<HTMLHeadingElement>;

export type AlertDescriptionProps = React.HTMLAttributes<HTMLDivElement>;
