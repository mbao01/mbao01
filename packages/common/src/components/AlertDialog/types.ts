import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";
import { type VariantProps } from "../../libs";
import { getButtonClasses } from "../Button/constants";
import {
  getDialogContentClasses,
  getDialogDescriptionClasses,
  getDialogFooterClasses,
  getDialogHeaderClasses,
  getDialogOverlayClasses,
  getDialogTitleClasses,
} from "../Dialog/constants";

export type AlertDialogProps = React.ComponentPropsWithoutRef<typeof AlertDialogPrimitive.Root>;

export type AlertDialogOverlayProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Overlay
> &
  VariantProps<typeof getDialogOverlayClasses>;

export type AlertDialogContentProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Content
> &
  VariantProps<typeof getDialogContentClasses>;

export type AlertDialogHeaderProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogHeaderClasses>;

export type AlertDialogFooterProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogFooterClasses>;

export type AlertDialogTitleProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Title
> &
  VariantProps<typeof getDialogTitleClasses>;

export type AlertDialogDescriptionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Description
> &
  VariantProps<typeof getDialogDescriptionClasses>;

export type AlertDialogActionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;

export type AlertDialogCancelProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;
