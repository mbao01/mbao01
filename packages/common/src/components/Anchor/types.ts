import { VariantProps } from "../../libs";
import { getAnchorClasses } from "./constant";

export type AnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & {
  isExternal?: boolean;
} & VariantProps<typeof getAnchorClasses>;
