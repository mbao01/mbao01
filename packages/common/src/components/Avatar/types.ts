import * as AvatarPrimitive from "@radix-ui/react-avatar";
import { type VariantProps } from "../../libs";
import {
  getAvatarClasses,
  getAvatarFallbackClasses,
  getAvatarImageClasses,
} from "./constants";

export type AvatarProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>,
  "asChild" | "size"
> &
  VariantProps<typeof getAvatarClasses>;

export type AvatarImageProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>,
  "asChild"
> &
  VariantProps<typeof getAvatarImageClasses>;

export type AvatarFallbackProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>,
  "asChild"
> &
  VariantProps<typeof getAvatarFallbackClasses>;
