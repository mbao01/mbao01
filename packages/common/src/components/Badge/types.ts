import { type VariantProps } from "../../helpers";
import { getBadgeClasses } from "./constants";

export type BadgeProps = React.HTMLAttributes<HTMLSpanElement> &
  VariantProps<typeof getBadgeClasses>;
