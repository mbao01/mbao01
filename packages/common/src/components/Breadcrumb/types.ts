import { ReactNode } from "react";

export type BreadcrumbProps = React.HTMLAttributes<HTMLElement> & {
  separator?: ReactNode;
};

export type BreadcrumbListProps = React.HTMLAttributes<HTMLUListElement>;

export type BreadcrumbItemProps = React.HTMLAttributes<HTMLLIElement>;

export type BreadcrumbLinkProps = React.HTMLAttributes<HTMLAnchorElement> & {
  asChild?: boolean;
};

export type BreadcrumbPageProps = React.HTMLAttributes<HTMLSpanElement>;

export type BreadcrumbSeparatorProps = React.HTMLAttributes<HTMLLIElement>;

export type BreadcrumbEllipsisProps = React.HTMLAttributes<HTMLSpanElement>;
