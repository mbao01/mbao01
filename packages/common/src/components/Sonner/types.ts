import { Toaster } from "sonner";
import { type VariantProps } from "../../libs";
import { getToastButtonClasses, getToastClasses } from "./constants";

type ToastButtonVariantProps = VariantProps<typeof getToastButtonClasses>;

export type ToasterProps = Omit<React.ComponentProps<typeof Toaster>, "closeButton"> &
  VariantProps<typeof getToastClasses> &
  Partial<Record<"actionButton" | "cancelButton" | "closeButton", ToastButtonVariantProps>> & {
    showCloseButton?: boolean;
  };
