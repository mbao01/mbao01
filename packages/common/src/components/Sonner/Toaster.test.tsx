import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "../Button";
import { toast, Toaster } from "./";
import { type ToasterProps } from "./types";

describe("Toaster", () => {
  const renderToaster = ({ onClick, ...props }: ToasterProps & { onClick: () => void }) =>
    render(
      <div>
        <Button onClick={onClick}>Show toast</Button>
        <Toaster {...props} />
      </div>
    );

  it("renders a basic toast", async () => {
    const user = userEvent.setup();
    renderToaster({
      onClick: () => toast("A simple toast"),
    });

    const button = screen.getByRole("button", { name: "Show toast" });

    await user.click(button);

    expect(screen.getByText("A simple toast")).toBeInTheDocument();
  });

  it.each([
    ["has outline", true],
    ["has no outline", false],
  ] as const)("%s", async (description, outline) => {
    const user = userEvent.setup();
    renderToaster({
      outline,
      onClick: () => toast.success("Sent successfully!", { description }),
    });

    const button = screen.getByRole("button", { name: "Show toast" });

    await user.click(button);

    expect(screen.getByText("Sent successfully!")).toBeInTheDocument();
  });

  it.each([
    "neutral",
    "primary",
    "secondary",
    "accent",
    "ghost",
    "info",
    "success",
    "warning",
    "error",
  ] as const)("has %s variant", async (variant) => {
    const heading = `Toaster ${variant}`;
    const user = userEvent.setup();
    renderToaster({
      variant,
      onClick: () =>
        toast(heading, {
          description: "This is a basic toast",
        }),
    });

    const button = screen.getByRole("button", { name: "Show toast" });

    await user.click(button);

    expect(screen.getByText(heading)).toBeInTheDocument();
    expect(screen.getByText("This is a basic toast")).toBeInTheDocument();
  });
});
