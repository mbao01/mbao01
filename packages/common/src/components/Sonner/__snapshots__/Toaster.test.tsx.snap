// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toaster > has accent variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-accent text-accent-content border-accent"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster accent
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has error variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-error text-error-content border-error"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster error
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has ghost variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-ghost text-ghost-content border-transparent"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster ghost
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has info variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-info text-info-content border-info"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster info
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has neutral variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-neutral text-neutral-content border-neutral"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster neutral
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has no outline 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-default text-default-content border-default"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-type="success"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-icon=""
          >
            <svg
              fill="currentColor"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Sent successfully!
            </div>
            <div
              class=""
              data-description=""
            >
              has no outline
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has outline 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group border-default border bg-default text-default"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-type="success"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-icon=""
          >
            <svg
              fill="currentColor"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Sent successfully!
            </div>
            <div
              class=""
              data-description=""
            >
              has outline
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has primary variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-primary text-primary-content border-primary"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster primary
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has secondary variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-secondary text-secondary-content border-secondary"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster secondary
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has success variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-success text-success-content border-success"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster success
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > has warning variant 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-warning text-warning-content border-warning"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              Toaster warning
            </div>
            <div
              class=""
              data-description=""
            >
              This is a basic toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Toaster > renders a basic toast 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn min-h-fit h-10"
    >
      Show toast
    </button>
    <section
      aria-label="Notifications alt+T"
      tabindex="-1"
    >
      <ol
        data-sonner-toaster="true"
        data-theme="light"
        data-x-position="right"
        data-y-position="bottom"
        dir=""
        style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
        tabindex="-1"
      >
        <li
          aria-atomic="true"
          aria-live="polite"
          class="sonner group bg-default text-default-content border-default"
          data-dismissible="true"
          data-expanded="false"
          data-front="true"
          data-index="0"
          data-mounted="true"
          data-promise="false"
          data-removed="false"
          data-sonner-toast=""
          data-styled="true"
          data-swipe-out="false"
          data-swiping="false"
          data-visible="true"
          data-x-position="right"
          data-y-position="bottom"
          role="status"
          style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
          tabindex="0"
        >
          <div
            class=""
            data-content=""
          >
            <div
              class=""
              data-title=""
            >
              A simple toast
            </div>
          </div>
        </li>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;
